core/src/main/java/com/novoda/merlin/service/MerlinService.java

```diff
@@ -65,6 +68,14 @@ void setConnectivityChangesForwarder(ConnectivityChangesForwarder connectivityCh
         }
 
         void onBindComplete() {
+            if (MerlinService.this.connectivityChangesRegister == null) {
+                throw new IllegalStateException("setConnectivityChangesRegister must be called.");
```

Comment:
I could move the `isBound = true` here because the `bind` is only successful if this is done too. Not sure if this will have an effect on the `MerlinServiceBinder` though 🤔